js/data.js

```diff
@@ -7,13 +7,65 @@ let data = {
   nextEntryId: 1,
 };
 
-const previousData = localStorage.getItem('javascript-local-storage');
-
 window.addEventListener('beforeunload', (event) => {
   const dataModel = JSON.stringify(data);
-  localStorage.setItem('javascript-local-storage', dataModel)
+  localStorage.setItem('javascript-local-storage', dataModel);
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  const previousData = localStorage.getItem('javascript-local-storage');
+
+  if (previousData !== null) {
+    data = JSON.parse(previousData);
+    data.entries.forEach((entry) => {
+      renderEntry(entry);
+    });
+    viewSwap(data.view);
+      if (data.entries.length > 0) {
+        toggleNoEntries();
+      }
+  } else {
+    viewSwap(data.view)
+  }
 });
 
-if (previousData !== null) {
-  data = JSON.parse(previousData);
+function renderEntry(entry) {
```

Comment:
the logic for `renderEntry` should be in your `main.js` file. 

Just keep your `data.js` file to load/unload local storage data. All the view swapping, event listeners, rendering logic, etc should be in `main.js`